File: python/Singleton/Singleton.py
class Singleton:

    _INSTANCE = None

    class __Singleton:
        def __init__(self):
            self._number = None
        
        def setNumber(self, number):
            self._number = number
        
        def getNumber(self):
            return self._number
    
    def __new__(self):
        if not Singleton._INSTANCE:
            Singleton._INSTANCE = Singleton.__Singleton()
        
        return Singleton._INSTANCE
    
    def __getattr__(self, name):
        return getattr(self._INSTANCE, name)

File: typescript/Singleton/src/singleton/singleton.ts
export class Singleton {

    private static INSTANCE: Singleton;

    private _number: number;

    private constructor() {
        this._number = 0;
    }

    public static getInstance(): Singleton {

        if (!Singleton.INSTANCE) {
            Singleton.INSTANCE = new Singleton();
        }

        return Singleton.INSTANCE;
    }

    public setNumber(n: number): void {
        this._number = n;
    }

    public getNumber(): number {
        return this._number;
    }

}
